Failure : "integer overflow in exp" on Altera-Max 10 platform

Test zephyr/tests/kernel/init fails to build and throws an error " integer overflow in exp " .

Configuration : 
<Arch> nios2
<Board> Altera Max 10

Commit : 8ba0ddf4baeb967a03ea36bb61b150bf3890adaf

Error description : 
 zephyr/tests/kernel/init/src/main.c: In function ‘verify_bootdelay’:
build output: zephyr/tests/kernel/init/src/main.c:26:43: error: integer overflow in exp
build output:         (CONFIG_BOOT_DELAY * NSEC_PER_USEC * USEC_PER_MSEC),
build output:                                            ^
build output: zephyr/tests/ztest/include/ztest_assert.h:92:11: note: in definition of m
build output:   _zassert(cond, msg ? ("(" default_msg ")") : (default_msg), \
build output:            ^~~~
build output: zephyr/tests/kernel/init/src/main.c:25:2: note: in expansion of macro ‘za
build output:   zassert_true(SYS_CLOCK_HW_CYCLES_TO_NS64(current_cycles) >=
build output:   ^~~~~~~~~~~~

Afaik yes (8b3fd6963c1b382660eac0bfdc5869dedcf745f1); setting the boot delay to 3000 (three seconds), I got an slightly different -- but same error for NIos2:

```
zephyr.git/tests/kernel/init/src/main.c:25:95: error: integer overflow in expression [-Werror=overflow] zassert_true(SYS_CLOCK_HW_CYCLES_TO_NS64(current_cycles) >=
cc1: all warnings being treated as errors
```

--- a/tests/kernel/init/src/main.c
+++ b/tests/kernel/init/src/main.c
@@ -6,6 +6,7 @@
 
 #include <ztest.h>
 
+#define NSEC_PER_MSEC (u64_t)(NSEC_PER_USEC * USEC_PER_MSEC)
 /**
  * @brief Test delay during boot
  * @defgroup kernel_bootdelay_tests Init
@@ -23,8 +24,8 @@ void verify_bootdelay(void)
 
 	/* compare this with the boot delay specified */
 	zassert_true(SYS_CLOCK_HW_CYCLES_TO_NS64(current_cycles) >=
-		     (CONFIG_BOOT_DELAY * NSEC_PER_USEC * USEC_PER_MSEC),
-		     "boot delay not executed");
+			(NSEC_PER_MSEC * CONFIG_BOOT_DELAY),
+			"boot delay not executed");
 }
 
 /**

